src/main/java/org/bihealth/mi/easybus/implementations/email/ConnectionSettingsIMAP.java

```diff
@@ -824,47 +759,66 @@ public static ConnectionIMAPSettings getConnectionIMAPSettingsFromFile(File file
-                System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + SMTP_PORT_KEY) == null ||
-                System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + SMTP_ENCRYPTION_TYPE) == null) {
-            throw new IllegalStateException("Properties file does not contain all necessary fields!");
-        }
-        if (!(System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + SMTP_ENCRYPTION_TYPE).equals(SSL_TLS) ||
-              System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + SMTP_ENCRYPTION_TYPE).equals(START_TLS)) ||
-            !(System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + IMAP_ENCRYPTION_TYPE).equals(SSL_TLS) ||
-              System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + IMAP_ENCRYPTION_TYPE).equals(SSL_TLS))) {
-            throw new IllegalStateException(String.format("Please set eithr %s or %s for IMAP and SMTP encryption", SSL_TLS, START_TLS));
-        }
-        
-        // Return
-        return new ConnectionIMAPSettings(System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + EMAIL_ADDRESS_KEY), null)
-                .setIMAPPassword(System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + PASSWORD_KEY))
-                .setIMAPServer(System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + IMAP_SERVER_KEY))
-                .setIMAPPort(Integer.valueOf(System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + IMAP_PORT_KEY)))
-                .setSMTPServer(System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + SMTP_SERVER_KEY))
-                .setSMTPPort(Integer.valueOf(System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + SMTP_PORT_KEY)))
-                .setSSLTLSIMAP(Boolean.valueOf(System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + IMAP_ENCRYPTION_TYPE).equals(SSL_TLS)))
-                .setSSLTLSSMTP(Boolean.valueOf(System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + SMTP_ENCRYPTION_TYPE).equals(SSL_TLS)))
-                .setAcceptSelfSignedCertificates(System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + ACCEPT_SELF_SIGNED_CERT_KEY) != null ? Boolean.valueOf(System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + ACCEPT_SELF_SIGNED_CERT_KEY)) : false)
-                .setSearchForProxy(System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + USE_PROXY_KEY) != null ? Boolean.valueOf(System.getProperties().getProperty(PREFIX_SYSTEM_PROPERTIES + USE_PROXY_KEY)) : false);
-    }
-    
     @Override
     public String toString() {
         return String.format("IMAP connections for e-mail address %s", this.imapEmailAddress);
     }
+
+    @Override
+    public String getIdentifier() {
+        return this.imapEmailAddress;
+    }
+
+    @Override
+    public ConnectionTypes getConnectionType() {
+        return ConnectionTypes.EMAIL;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(acceptSelfSignedCert,
```

Comment:
How has this method been generated / implemented? Why did you not use the mechanism for generating hashCode and equals provided by eclipse?

Reply:
I can not follow. it was generated with the named mechanism of eclipse